### PR TITLE
[docs] Add tldr section for coredb contributors

### PR DESCRIPTION
**Pull request by jaki**
CoreDB contributor docs for compiling have separate pieces of
instructions with some beneficial commentary, but there is no fast way
to do it in one shot.

Add a tldr section for all the linux platforms that collects all the
separate instructions into one place.  This means future edits to the
instructions may need to worry about keeping the two in sync, but the
convenience seems worth it.  Tested using docker images centos:7,
ubuntu:20.04, ubuntu:22.04, almalinux/8-minimal.  It would be nice to
create docker files using those images as a followup.

### CODE REVIEW

File: docs/content/preview/contribute/core-database/includes/tldr.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+The following assumes
```
Comment (aishwarya24) [inline docs/content/preview/contribute/core-database/includes/tldr.md:1]:
```suggestion
The following assumes that:

- you have sudo access
- you are using a bash or zsh shell
- you are running commands as the user who will be doing development.
```

**Review by aishwarya24 — APPROVED** [review]
LGTM with a minor suggestion

File: docs/content/preview/contribute/core-database/build-from-src-almalinux.md [hunk 2]
```diff
@@ -50,6 +50,53 @@ AlmaLinux 8 is the recommended Linux development platform for YugabyteDB.
...
+  npm
+  patchelf
+  python39
+  rsync
+)
+sudo dnf -y install "${packages[@]}"
+sudo alternatives --set python3 /usr/bin/python3.9
+latest_zip_url=$(curl -Ls "https://api.github.com/repos/ninja-build/ninja/releases/latest" \
+                 | grep browser_download_url | grep ninja-linux.zip | cut -d \" -f 4)
+curl -Ls "$latest_zip_url" | zcat | sudo tee /usr/local/bin/ninja >/dev/null
```
[the remark this answers is not in the document]
Comment (jaki) [reply docs/content/preview/contribute/core-database/build-from-src-almalinux.md:82]: -s is used.  sudo tee is piping to /dev/null

File: docs/content/preview/contribute/core-database/build-from-src-ubuntu.md [hunk 3]
```diff
@@ -52,6 +52,87 @@ AlmaLinux 8 is the recommended Linux development platform for YugabyteDB.
...
+         | tee -a "$shellrc")
+source <(echo 'export PATH="/usr/lib/go-1.20/bin:$PATH"' \
+         | tee -a "$shellrc")
+source <(echo 'export YB_CCACHE_DIR="$HOME/.cache/yb_ccache"' \
+         | tee -a "$shellrc")
+
+git clone https://github.com/yugabyte/yugabyte-db
+cd yugabyte-db
+case "$VERSION_ID" in
+  20.04)
```
[the remark this answers is not in the document]
Comment (jaki) [reply docs/content/preview/contribute/core-database/build-from-src-ubuntu.md:125]: no point in removing it yet

File: docs/content/preview/contribute/core-database/build-from-src-almalinux.md [hunk 4]
```diff
@@ -50,6 +50,53 @@ AlmaLinux 8 is the recommended Linux development platform for YugabyteDB.
...
+  java-1.8.0-openjdk
+  libatomic
+  maven
+  npm
+  patchelf
+  python39
+  rsync
+)
+sudo dnf -y install "${packages[@]}"
+sudo alternatives --set python3 /usr/bin/python3.9
```
[the remark this answers is not in the document]
Comment (jaki) [reply docs/content/preview/contribute/core-database/build-from-src-almalinux.md:79]: There is no `python-310` or `python-311` package yet on Almalinux 8 as far as I can tell.  Also, this is based off of the python section below, so changing both copies of this could be considered out of scope of this PR.

File: docs/content/preview/contribute/core-database/build-from-src-almalinux.md [hunk 5]
```diff
@@ -50,6 +50,53 @@ AlmaLinux 8 is the recommended Linux development platform for YugabyteDB.
 
 {{< /note >}}
 
+# TLDR
```
Comment (ddhodge) [inline docs/content/preview/contribute/core-database/build-from-src-almalinux.md:53]:
```suggestion
## TLDR
```

File: docs/content/preview/contribute/core-database/build-from-src-almalinux.md [hunk 6]
```diff
@@ -50,6 +50,53 @@ AlmaLinux 8 is the recommended Linux development platform for YugabyteDB.
...
+sudo chown "$USER" /opt/yb-build
+source <(echo 'export YB_CCACHE_DIR="$HOME/.cache/yb_ccache"' \
+         | tee -a "$shellrc")
+
+git clone https://github.com/yugabyte/yugabyte-db
+cd yugabyte-db
+./yb_release
+```
+
+# Detailed instructions
```
Comment (ddhodge) [inline docs/content/preview/contribute/core-database/build-from-src-almalinux.md:98]:
```suggestion
## Detailed instructions
```

File: docs/content/preview/contribute/core-database/build-from-src-almalinux.md [hunk 7]
```diff
@@ -50,6 +50,53 @@ AlmaLinux 8 is the recommended Linux development platform for YugabyteDB.
...
+
 ## Install necessary packages
```
Comment (ddhodge) [inline docs/content/preview/contribute/core-database/build-from-src-almalinux.md:100]:
```suggestion
```

File: docs/content/preview/contribute/core-database/build-from-src-centos.md [hunk 8]
```diff
@@ -52,6 +52,57 @@ AlmaLinux 8 is the recommended Linux development platform for YugabyteDB.
 
 The following instructions are for CentOS 7.
 
+# TLDR
```
Comment (ddhodge) [inline docs/content/preview/contribute/core-database/build-from-src-centos.md:55]:
```suggestion
## TLDR
```

File: docs/content/preview/contribute/core-database/build-from-src-centos.md [hunk 9]
```diff
@@ -52,6 +52,57 @@ AlmaLinux 8 is the recommended Linux development platform for YugabyteDB.
...
+         | tee -a "$shellrc")
+source <(echo 'export YB_CCACHE_DIR="$HOME/.cache/yb_ccache"' \
+         | tee -a "$shellrc")
+
+git clone https://github.com/yugabyte/yugabyte-db
+cd yugabyte-db
+./yb_release
+```
+
+# Detailed instructions
```
Comment (ddhodge) [inline docs/content/preview/contribute/core-database/build-from-src-centos.md:104]:
```suggestion
## Detailed instructions
```

File: docs/content/preview/contribute/core-database/build-from-src-centos.md [hunk 10]
```diff
@@ -52,6 +52,57 @@ AlmaLinux 8 is the recommended Linux development platform for YugabyteDB.
...
+
 ## Install necessary packages
```
Comment (ddhodge) [inline docs/content/preview/contribute/core-database/build-from-src-centos.md:106]:
```suggestion
```

File: docs/content/preview/contribute/core-database/build-from-src-ubuntu.md [hunk 11]
```diff
@@ -52,6 +52,87 @@ AlmaLinux 8 is the recommended Linux development platform for YugabyteDB.
 
 The following instructions are for Ubuntu 20.04 and 22.04.
 
+# TLDR
```
Comment (ddhodge) [inline docs/content/preview/contribute/core-database/build-from-src-ubuntu.md:55]:
```suggestion
## TLDR
```

File: docs/content/preview/contribute/core-database/build-from-src-ubuntu.md [hunk 12]
```diff
@@ -52,6 +52,87 @@ AlmaLinux 8 is the recommended Linux development platform for YugabyteDB.
...
+  20.04)
+    ./yb_release --build_args=--clang16
+    ;;
+  22.04)
+    ./yb_release --build_args=--clang17
+    ;;
+esac
+```
+
+# Detailed instructions
```
Comment (ddhodge) [inline docs/content/preview/contribute/core-database/build-from-src-ubuntu.md:134]:
```suggestion
## Detailed instructions
```

File: docs/content/preview/contribute/core-database/build-from-src-ubuntu.md [hunk 13]
```diff
@@ -52,6 +52,87 @@ AlmaLinux 8 is the recommended Linux development platform for YugabyteDB.
...
+
 ## Install necessary packages
```
Comment (ddhodge) [inline docs/content/preview/contribute/core-database/build-from-src-ubuntu.md:136]:
```suggestion
```

File: docs/content/preview/contribute/core-database/build-from-src-almalinux.md (hunk 5 above)
Comment (ddhodge) [inline docs/content/preview/contribute/core-database/build-from-src-almalinux.md:53]: In page headings start at level 2, that's what the styles support - bump all the headings down a level, or I provided an alternative
